addons/io/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java

```diff
@@ -330,7 +343,14 @@ private void handleRequestEvent(JSONObject data) {
             logger.debug("Request method is {}", requestMethod);
             Request request = jettyClient.newRequest(requestUri);
             setRequestHeaders(request, requestHeadersJson);
-            request.header("X-Forwarded-Proto", "https");
+            if (protocol != null) {
+                request.header("X-Forwarded-Proto", protocol);
```

Comment:
You could consider not duplicating the `request.header("X-Forwarded-Proto"` 3 times, but just collecting a string and passing that to the `request.header("X-Forwarded-Proto"`